repository/repository.go

```diff
@@ -89,6 +89,15 @@ func (r *Repository) Path() string {
 	return path
 }
 
+// PushPath gives us sanitized repository path and checks if subdirectories are allowed
```

Comment:
This functions does not actually check anything 🤔 Maybe

> // PushPath gives us repository path having custom path element separator

... or something like this. 

WDYT?